### PR TITLE
Adding feature routemap

### PR DESCRIPTION
**Pull request by websitescenes**

### CODE REVIEW

File: lib/rbeapi/api/routemaps.rb [hunk 1]
```diff
@@ -43,63 +43,439 @@ module Api
...
+          response[name] = parse_entries(entries)
+        end
+        response
+      end
+
+      ##
+      # entries regex scans the config for the specified route-name. The method
+      #   reduces duplication.
+      #
+      def entries_regex(name)
```
Comment (devrobo) [inline lib/rbeapi/api/routemaps.rb]: I think in all the other modules we using the naming convention parse_ , for this case parse_entry would work.

Comment (devrobo) [inline lib/rbeapi/api/routemaps.rb]: Why isn't this code in parse_entries? The pattern you have is:
entries = entries_regex(name)
parse_entries(entries)

File: lib/rbeapi/api/routemaps.rb [hunk 2]
```diff
@@ -43,63 +43,439 @@ module Api
...
-              hsh['continue_rules'] << rule.strip
+          rule_hsh = { action: mdata[2], seqno: mdata[3].to_i }
+          unless rules.nil?
+            rules.split("\n").each_with_object({}) do |rule|
+              parse_rule(rule, rule_hsh)
             end
           end
-          rule_hsh.update(parsed)
-          arry << rule_hsh
+          return rule_hsh
```
Comment (devrobo) [inline lib/rbeapi/api/routemaps.rb]: Why are you returning inside a loop?

File: lib/rbeapi/api/routemaps.rb [hunk 3]
```diff
@@ -43,63 +43,439 @@ module Api
...
+      end
+      private :parse_rule
+
+      ##
+      # name_commands is utilized by create to prepare the specified
+      # routemap.
+      #
+      def name_commands(name, opts)
+        if opts[:enable] == false
+          @cmd = "no route-map #{name}"
```
Comment (devrobo) [inline lib/rbeapi/api/routemaps.rb]: Why global variables?

File: lib/rbeapi/api/routemaps.rb (hunk 2 above)
Comment (websitescenes) [inline lib/rbeapi/api/routemaps.rb]: rule_hsh is only available in the loop. I can think about another way to write this if you don't like it.

File: lib/rbeapi/api/routemaps.rb (hunk 3 above)
Comment (websitescenes) [inline lib/rbeapi/api/routemaps.rb]: I used a global to pass @cmd from name_commands to parse_entries. I could also just return cmd without using a variable if you prefer.

File: lib/rbeapi/api/routemaps.rb (hunk 1 above)
Comment (websitescenes) [inline lib/rbeapi/api/routemaps.rb]: Good catch. I'll combine them.

**Comment by jerearista** [comment]
retest this please

**Comment by devrobo** [comment]
Retest this please.

**Pull request merged**